simulator/simulator.go

```diff
@@ -491,7 +491,23 @@ func (s *Server) Close() {
 	}
 }
 
-var typeFunc = types.TypeFunc()
+var (
+	vim25MapType = types.TypeFunc()
+	typeFunc     = defaultMapType
+)
+
+func defaultMapType(name string) (reflect.Type, bool) {
+	typ, ok := vim25MapType(name)
+	if !ok {
+		// See TestIssue945, in which case Go does not resolve the namespace and name == "ns1:TraversalSpec"
+		// Without this hack, the SelectSet would be all nil's
```

Comment:
need to take a closer look. feels hacky. seems like we are trimming only `vim25:` as namespace here: 

https://github.com/vmware/govmomi/blob/5072cda664c79ada30834d171d2ed1f76317d3b2/vim25/types/registry.go#L38

Reply:
It is a hack, like the comment says.  We've only hit this with pyvsphere (not pyvmomi or other clients).  I don't want to spend time messing with our fork of the xml package at the moment.

Reply:
not the xml fork. just `TypeFunc` in `types/registry.go`. If you replace the `TrimPrefix` with `SplitN`, it should work as well. or maybe we can have a `TypeFunc` that only trims vim25, and anther that trims all namespaces.

Reply:
Yeah I didn't want to put the hack in the types package.  The vim25 check there is valid.  But the hack above we shouldn't need at all, as the xml package should be properly resolving the namespace alias.

Reply:
oh, i see, there is this: `<SOAP-ENV:Body xmlns:ns1="urn:vim25">` missed that.